src/App.js

```diff
@@ -1,13 +1,105 @@
 import React, { Component } from 'react'
+import PropTypes from 'prop-types'
 import { Instructions } from './components'
 import './styles/app.css'
 
+class Counter extends Component {
+  static propTypes = {
+    initialCount: PropTypes.number.isRequired
+  }
+  constructor(props) {
+    super(props)
+    this.state = { count: this.props.initialCount }
+  }
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      count: nextProps.initialCount
+    })
+  }
+  incClick = () => {
+    this.setState(state => ({
+      count: state.count + 1
+    }))
+  }
+  decClick = () => {
+    this.setState(state => ({
+      count: state.count - 1
+    }))
+  }
+  render() {
+    return (
+      <div>
+        Count state: {this.state.count}
+        <button onClick={this.decClick}>Decrease</button>
+        <button onClick={this.incClick}>Increase</button>
+      </div>
+    )
+  }
+}
+
+class InitialCountForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      count: 0,
+      newCount: 0
+    }
+  }
+  handleSubmit(event) {
+    this.setState({
+      count: Number(this.state.newCount),
+      newCount: this.state.newCount
+    })
+    this.props.appCallback(Number(this.state.newCount))
+  }
+  handleForm(event) {
+    this.setState({
+      count: Number(this.state.count),
+      newCount: event.target.value
+    })
+  }
+  render() {
+    return (
+      <div>
+        <label>
+          Set count state:
+          <input
```

Comment:
Since this is a "controlled" component, you should include the attribute of `value={this.state.newCount}`
Here's some good reading: https://reactjs.org/docs/forms.html#controlled-components